app/services/tasks/samples_to_repository_migration_service.rb

```diff
@@ -151,5 +182,121 @@ def self.get_custom_columns(team, repository)
       prepare_text_value_custom_columns(team, repository) +
         prepare_list_value_custom_columns_with_list_items(team, repository)
     end
+
+    def self.create_text_cell(row_id, column_id, data,
+                              created_by_id, last_modified_by_id)
+      conn = ActiveRecord::Base.connection
+      timestamp = conn.quote(Time.now.to_s(:db))
+      values = [conn.quote(data), created_by_id, last_modified_by_id, timestamp, timestamp]
+      value_sql = <<-SQL
+        INSERT INTO repository_text_values
+          (data, created_by_id, last_modified_by_id,
+           created_at, updated_at)
+        VALUES (#{values.join(', ')})
+        RETURNING id
+      SQL
+
+      value_id = conn.execute(value_sql)[0]['id']
+
+      values = [row_id, column_id, value_id, conn.quote('RepositoryTextValue'),
+                timestamp, timestamp]
+      cell_sql = <<-SQL
+        INSERT INTO repository_cells
+          (repository_row_id, repository_column_id, value_id, value_type,
+           created_at, updated_at)
+        VALUES (#{values.join(', ')})
+      SQL
+      conn.execute(cell_sql)
+    end
+
+    def self.create_list_cell(row_id, column_id, list_item_id,
+                              created_by_id, last_modified_by_id)
+      conn = ActiveRecord::Base.connection
+      timestamp = conn.quote(Time.now.to_s(:db))
+      values = [list_item_id, created_by_id, last_modified_by_id,
+                timestamp, timestamp]
+      list_value_sql = <<-SQL
+        INSERT INTO repository_list_values
+          (repository_list_item_id, created_by_id, last_modified_by_id,
+           created_at, updated_at)
+        VALUES (#{values.join(', ')})
+        RETURNING id
+      SQL
+      value_id = ActiveRecord::Base.connection.execute(list_value_sql)[0]['id']
+
+      values = [row_id, column_id, value_id, conn.quote('RepositoryListValue'),
+                timestamp, timestamp]
+      cell_sql = <<-SQL
+        INSERT INTO repository_cells
+          (repository_row_id, repository_column_id, value_id, value_type,
+           created_at, updated_at)
+        VALUES (#{values.join(', ')})
+      SQL
+      ActiveRecord::Base.connection.execute(cell_sql)
+    end
+
+    def self.update_smart_annotations(team, mappings)
```

Comment:
Do we need to also update existing reports that have smart annotations inside?

Reply:
I think reports pull actual data from db